tests/memoryview/pure_optional.pyx

```diff
@@ -0,0 +1,13 @@
+# mode: run
+# tag: pure, memoryview
+import cython
+
+
+def slice_optional(m: cython.double[:]):
```

Comment:
I think this could be in memoryview_annotation_typing probably (and probably be called `slice_not_optional`)

I'm just aware that the memoryview tests are reasonably slow (because memoryviews generate a big chunk of utility code) so trying to keep the number of test files down if possible

Reply:
> I think this could be in memoryview_annotation_typing probably (and probably be called slice_not_optional)

I was trying to put it there in the first place but I the problem was that .py tests suite is run for both as compiled and not compiled. Normally it is not a problem but in case of this test, the expected result behave differently when run in Cython and in Python. But I have an Idea how to fix it.

Reply:
Moved. Not sure whether using `cython.compiled()` in tests is OK or not.

Reply:
Don't see the problem with using `compiled` in this case